dataactbroker/README.md

```diff
@@ -240,67 +240,84 @@ Example Output:
+A call to this route should be of content type `"multipart/form-data"`, and should use @ notation for the values of the "appropriations", "program_activity" and "award_financial" keys, to indicate the local path to the files to be uploaded.
 
-When upload is complete, the finalize_submission route should be called with the job_id.
+If using the frontend and passing filename strings rather than files, this route will return conflict free S3 URLs for uploading. Each key put in the request comes back with a url_key containing the S3 URL and a key_id containing the job id. A returning submission_id will also exist which acts as identifier for the submission.
 
-Example input:
+In addition, with frontend use, a credentials object is also part of the returning request. This object provides temporary access to upload S3 Files using an AWS SDK. It contains the following: SecretAccessKey, SessionToken, Expiration, and AccessKeyId. It is important to note that the role used to create the credentials should be limited to just S3 access.
 
-```json
-{
-  "appropriations":"appropriations.csv",
-  "award_financial":"award_financial.csv",
-  "award":"award.csv",
-  "program_activity":"program_activity.csv",
-  "agency_name":"Name of the agency",
-  "reporting_period_start_date":"03/31/2016",
-  "reporting_period_end_date":"03/31/2016",
-  "is_quarter":False,
-  "existing_submission_id: 7 (leave out if not correcting an existing submission)
-}
-```
+This route will upload the files, then kick off the validation jobs. It will return the submission id, which can be used for the `/v1/check_status/` route to poll for validation completion. 
 
-Example output:
+**NOTE: If using the frontend, you will need to call /v1/finalize_job/ to kick off validation once upload is complete**
 
-```json
-{
-  "submission_id": 12345,
 
-  "bucket_name": "S3-bucket",
+#### Additional Required Headers:
 
-  "award_id": 100,
-  "award_key": "2/1453474323_awards.csv",
+- `Content-Type` - `"multipart/form-data"`
 
-  "appropriations_id": 101,
-  "appropriations_key": "2/1453474324_appropriations.csv",
+#### Request Parameters:
+- `appropriations` - local path to file using @ notation
+- `program_activity` - local path to file using @ notation
+- `award_financial` - local path to file using @ notation
+- `cgac_code` - **required if not FREC** string, CGAC of agency (null if FREC agency)
+- `frec_code` - **required if not CGAC** string, FREC of agency (null if CGAC agency)
+- `is_quarter` - boolean (true for quarterly submissions)
+- `reporting_period_start_date` - string, starting date of submission (MM/YYYY)
+- `reporting_period_end_date` - string, ending date of submission (MM/YYYY)
 
-  "award_financial_id": 102,
-  "award_financial_key": "2/1453474327_award_financial.csv",
+**NOTE**: for monthly submissions, start/end date are the same
 
-  "program_activity_id": 103,
-  "program_activity_key": "2/1453474333_program_activity.csv",
+#### Example curl request:
```

Comment:
We'll also want an `Example Request` here, which should be an example of what the front end is submitting.

Reply:
Done